src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectSystem/RazorProjectService.cs

```diff
@@ -285,12 +284,6 @@ await _projectManager.UpdateAsync(
                     filePath,
                     (project, textDocumentPath) =>
                     {
-                        if (!project.DocumentFilePaths.Contains(textDocumentPath, FilePathComparer.Instance))
```

Comment:
>if (!project.DocumentFilePaths.Contains(textDocumentPath, FilePathComparer.Instance)) [](http://example.com/codeflow?start=24&length=85)

This doesn't look like it belongs

Reply:
Because it's redundant to call `ContainsDocument(...)` and immediately call `TryGetDocument(...)` with the same document file path. It would take a lock and perform the same dictionary lookup twice. If `ContainsDocument(...)` succeeds, so will `TryGetDocument(...)`.